lib/ghtorrent/adapters/mongo_persister.rb

```diff
@@ -60,6 +60,12 @@ def store(entity, data = {})
       mongo[entity].insert_one(data).to_s
     end
 
+    def replace(entity, query, new_entry, upsert = true)
```

Comment:
In the [original code](https://github.com/gousiosg/github-mirror/blob/6082fe4fe5d223b28be83cb1a7bed8fc53e49bd3/lib/ghtorrent/adapters/mongo_persister.rb#L84), the `upsert` method performs a "delete" and a "store". This will not generate a Mongo DB Update event, which is required to generate the corresponding event on Rabbit MQ to indicate that there are new data for a repository.

We had a design philosophy to change the currently working code *as little as possible*. Therefore, we chose to implement new functionality rather than change existing functionality.

This philosophy is what led to the creation of the [replace method](https://github.com/gousiosg/github-mirror/blob/ee9d46dd7917515564139e713bedf6dbaf344c7d/lib/ghtorrent/adapters/mongo_persister.rb#L63).

So we see at least two possible paths forward:

1. Leave the new functionality and continue to differentiate between the `upsert` functionality (delete and insert), which will generate "insert" events and the `replace` functionality, which will generate "update" events.
2. Replace the current `upsert` functionality with the `replace` implementation, which will:
2.1 DRY up the code
2.2 Generate more MongoDB "update" events. We do not believe that having more "update" events instead of "insert" events would be a concern.

Reply:
I agree that the code in your current `replace` method makes much more sense. I would follow your option 2, so delete your `replace` and copy its code in my `upsert`.